python/e2b_code_interpreter/code_interpreter_async.py

```diff
@@ -191,6 +191,10 @@ async def run_code(
         request_timeout = request_timeout or self.connection_config.request_timeout
         context_id = context.id if context else None
 
+        headers = {}
+        if self._envd_access_token:
```

Comment:
if the sandbox isn't secure, this will be `None` and the request fails on parsing (this is handled correctly in JS already)